src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -16917,6 +16917,11 @@ void Unit::_EnterVehicle(Vehicle* vehicle, int8 seatId, AuraApplication const* a
         }
     }
 
+    // For vehicles, check if there is unit type non move
+    // If so, root the vehicle once someone gets in
+    if (HasFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_DISABLE_MOVE))
```

Comment:
Looking at this more. So far as I can see the player inherits the unit state. Hence why this works fine. All I think should be done is correct the comment.

I did originally read from the template.. But, I think the problem there is if somehow the root is legitimately removed, this would break that.

Checking the root flag on the player seems to work fine though.